rpcs3/Emu/Cell/SPURecompiler.cpp

```diff
@@ -6521,6 +6521,23 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 		}
 		case SPU_RdDec:
 		{
+			if (utils::get_tsc_freq() && !(g_cfg.core.spu_loop_detection))
+			{
+				const auto timestamp = m_ir->CreateLoad(get_type<u64>(), spu_ptr<u64>(&spu_thread::ch_dec_start_timestamp));
+				const auto dec_value = m_ir->CreateLoad(get_type<u32>(), spu_ptr<u32>(&spu_thread::ch_dec_value));
+				const auto tsc = m_ir->CreateCall(get_intrinsic(llvm::Intrinsic::x86_rdtsc));
+				const auto tscns = m_ir->CreateUDiv(m_ir->CreateMul(tsc, m_ir->getInt64(utils::get_tsc_freq())), m_ir->getInt64(1000000000));
```

Comment:
1. Needs check against timer scaling not be 100%
2. SPU loop detection setting need to be marked as non-dynamic.
3. This direct multiplication then division may overflow, in get_system_time and get_timebased_time there is a better method for multiplying integers by ratio which guards against most overflows.